ft-sdk/src/in_.rs

```diff
@@ -16,4 +16,33 @@ impl In {
             form_errors: std::collections::HashMap::new(),
         })
     }
+
+    pub fn user_agent(&self) -> String {
+        self.req
+            .headers()
+            .get(http::header::USER_AGENT)
+            .and_then(|v| v.to_str().map(|v| v.to_string()).ok())
```

Comment:
Leave changes in this file, and update the PR title, we will merge this. Undo the Action2.